src/System.Windows.Forms/src/System/Windows/Forms/Button.ButtonAccessibleObject.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+using static System.Windows.Forms.Control;
+using static Interop;
+
+namespace System.Windows.Forms
+{
+    public partial class Button
+    {
+        [ComVisible(true)]
+        internal class ButtonAccessibleObject : ControlAccessibleObject
+        {
+            internal ButtonAccessibleObject(Button owner) : base(owner)
+            {
+            }
+
+            private Button OwningButton => Owner as Button;
+
+            internal override bool IsIAccessibleExSupported() => true;
+
+            internal override object GetPropertyValue(UiaCore.UIA propertyID)
+            {
+                switch (propertyID)
+                {
+                    case UiaCore.UIA.NamePropertyId:
+                        return Name;
+                    case UiaCore.UIA.AutomationIdPropertyId:
+                        return OwningButton.Name;
```

Comment:
OwningButton property getter is a good candidate to add check for `IsHandleCreated`.